README.md

```diff
@@ -20,14 +20,14 @@ i.e. for ``data`` tags that are direct children of the ``graph`` tag.
-| creationtime | *Valid values:* Date and time of creation in ISO 8601 format. <br /> The date and time the witness was created in ISO 8601 format. The date must contain the year, the month, and the day, separated by dashes ('-'). The date is separated from the time using the capital letter 'T'. The time must be given in hours, minutes, and seconds, separated by colons (':'). If the timestamp is not given in UTC time, a positive ('+') or negative ('-') time offset consisting of hours and minutes separated by a colon (':') can be appended. Example: ``2016-12-24T13:15:32+02:00``. | Yes |
+| witness-type | *Valid values:* ``correctness_witness`` or ``violation_witness`` <br /> ``witness-type`` is used to specify the witness type. A correctness witness is identified by the value ``correctness_witness``, a violation witness is identified by the value ``violation_witness``. | Yes |
+| sourcecodelang | *Valid values:* Currently, only ``C`` and ``Java`` are supported. <br /> ``sourcecodelang`` is used to specify the name of the programming language, for example ``C``. | Yes |
+| producer | *Valid values:* Any <br /> ``producer`` is used to specify the name of the tool that produced the witness automaton, for example ``CPAchecker 1.6.8`` | Yes |
+| specification | *Valid values:* The specification text <br /> ``specification`` is used to provide a textual representation of the specification of the verification task. The format of this representation is user defined. In SV-COMP, the text ``CHECK( init(main()), LTL(G ! call(__VERIFIER_error())) )`` is used to represent the specification. | Yes |
+| programfile | *Valid values:* The program file path as passed to the verifier <br /> ``programfile`` is used to record the URI or file-system path to the source code, e.g., ``loop-acceleration/multivar_true-unreach-call1_true-termination.i``. This key is intended for documentation purposes; a validator is not required to be able to access the specified file location, because the source code is explicitly provided to the validator as input. Hence, the validity of the witness must not depend on the availability of the source code at the location specified by the value of this key in the execution environment of the validation. | Yes |
+| programhash | *Valid values:* SHA-256 hash <br /> ``programhash`` is used to record the SHA-256 hash value of the verified program, for example, ``e2d5365a863c1c57fbe2870942676040efc3aea2d9bb085092800d6e256daf06``. | Yes |
+| architecture | *Valid values:* An identifier for the assumed architecture <br /> ``architecture`` is used to provide a textual representation of the machine architecture assumed for the verification task. This textual representation is user defined. For example, use the identifiers ``32bit`` and ``64bit`` to distinguish between 32-bit systems and 64-bit systems. | Yes |
```

Comment:
Yes, we can afford to leave this user defined. The field is mandatory for two reasons (off the top of my head):
- to force witness producers to think about what they are doing (architecture matters)
- documentation (which architecture did the producer assume when producing the witness?)
- reproducibility (same verifier may output different witnesses for different architectures it supports --- how do you tell them apart?)

I would welcome a more precise data model, but I suspect it would be tied to specific assumptions about programming languages and systems, whereas witnesses are *not* only for C/Java, ILP32, LP64 etc. and could also be used for hypothetical virtual systems that are not even designed yet. I suspect that for such cases, we will always need a verifier-defined / community-agreed architecture.